reference-architecture/day2ops/scripts/ocp36-sat6.py

```diff
@@ -0,0 +1,85 @@
+
+    def __init__(self, load=True):
+
+        if load:
+            self._parseCli()
+            self._loadImageList()
+            self._addData()
+            self._syncData()
+
+    def _loadImageList(self):
+        cmd='curl -s https://registry.access.redhat.com/v1/search?q="openshift3" | python -mjson.tool | grep ".name.:" | cut -d: -f2 | sed -e "s/ "//g"" -e "s/,"//g""'
+        result = subprocess.check_output(cmd, shell=True)
+        lines = result.splitlines()
+        for line in lines:
+            nl = line.replace('"', '')
+            self.openshift3Images.append(nl)
+
+    def _parseCli(self):
+
+        parser = argparse.ArgumentParser(description='Add all OCP images for disconnected installation to satellite 6', add_help=True)
+        parser.add_argument('--orgid', action='store', default='1',help='Satellite organization ID to create new product for OCP images in')
+        parser.add_argument('--productname', action='store', default='ocp36',help='Satellite product name to use to create OCP images')
+        parser.add_argument('--username', action='store', default='admin', help='Satellite 6 username for hammer CLI')
+        parser.add_argument('--password', action='store', help='Satellite 6 Password for hammer CLI')
+        parser.add_argument('--no_confirm', action='store_true', help='Satellite 6 Password for hammer CLI')
+        self.args = parser.parse_args()
+
+        if not self.args.password:
+            self.args.password = getpass.getpass(prompt='Please enter the password to use for the admin account in hammer CLI: ')
+
+    def _syncData(self):
+
+        if not self.args.no_confirm:
+            print "Sync repo data? (This may take a while)"
+            go = raw_input("Continue? y/n:\n")
+            if 'y' not in go:
+                exit(0)
+
+        cmd="hammer --username %s --password %s product synchronize --name %s --organization-id %s" % (self.args.username, self.args.password, self.args.productname, self.args.orgid)
+        os.system(cmd)
+
+    def _addData(self):
+
+        if not self.args.no_confirm:
+            print "Adding OCP images to org ID: %s with the product name: %s" % (self.args.orgid, self.args.productname)
+            go = raw_input("Continue? y/n:\n")
+            if 'y' not in go:
+                exit(0)
+
+        print "Adding product with name: %s" % self.args.productname
+
+        cmd="hammer --username %s --password %s product create --name %s --organization-id %s" % (self.args.username, self.args.password, self.args.productname, self.args.orgid)
+        os.system(cmd)
+
+        print "Adding openshift3 images"
+        for image in self.openshift3Images:
+            cmd='hammer --username %s --password %s repository create --name %s --organization-id %s --content-type docker --url "https://registry.access.redhat.com" --docker-upstream-name %s --product %s' % (self.args.username,  self.args.password, image, self.args.orgid, image, self.args.productname )
```

Comment:
This create a repo for every image. I have 0 idea about the docker stuff in satellite, but it won't be better to create a single repo with all the images? (As I said, I have 0 idea, so maybe I'm talking nonsense)

Reply:
You cannot do that. It's one repo per container image.